lib/oauth.js

```diff
@@ -240,8 +249,10 @@ module.exports = new Model({
   },
 
   _refreshBearerToken: function() {
+    this._deleteBearerToken();
```

Comment:
If we unset the `_currentUserPromise` like https://github.com/zooniverse/panoptes-javascript-client/blob/85bdc163aefaa3c84e9fac101408f590dfba8636/lib/oauth.js#L122 we may be able to signal to the calling app that the authentication renewal failed. 

If we can get the the calling app react to this state we can update the UI. I don't think we can do this in SW / Annotate via [auth factory](https://github.com/zooniverse/shakespeares_world/blob/1d52a27215960b594893e69c40dc57807cf527cc/app/modules/auth/auth.factory.js#L15) without a code change.

Reply:
That promise resolves when `checkCurrent()` resolves, as far as I can tell. Once a promise is resolved, further changes don't affect it. So, I think this line here cancels any calls to `checkCurrent()` that are in progress from the client app, by immediately resolving the session check with a null value.